libs/perun/components/src/lib/consents-list/consents-list.component.html

```diff
@@ -0,0 +1,126 @@
+          <mat-checkbox
+            color="primary"
+            (click)="$event.stopPropagation()"
+            (change)="$event ? selection.toggle(consent) : null"
+            [checked]="selection.isSelected(consent)">
+          </mat-checkbox>
+        </td>
+      </ng-container>
+
+      <ng-container matColumnDef="status">
+        <th *matHeaderCellDef mat-header-cell mat-sort-header>
+          {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.STATUS' | customTranslate | translate}}
+        </th>
+        <td mat-cell *matCellDef="let consent">
+          <perun-web-apps-consent-status
+            [consentStatus]="consent.status"></perun-web-apps-consent-status>
+        </td>
+      </ng-container>
+
+      <ng-container matColumnDef="name">
+        <th *matHeaderCellDef mat-header-cell mat-sort-header>
+          {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.SERVICE_NAME' | customTranslate | translate}}
+        </th>
+        <td mat-cell *matCellDef="let consent">
+          {{consent.consentHub.name}}
+        </td>
+      </ng-container>
+
+      <ng-container matColumnDef="expandedDetail">
+        <td mat-cell *matCellDef="let consent" [attr.colspan]="displayedColumns.length">
+          <div
+            [@detailExpand]="consent === expandedConsent ? 'expanded' : 'collapsed'"
+            class="consent-detail">
+            <div class="p-3">
+              <div *ngIf="consent.status === 'GRANTED'">
+                {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.GRANTED_TEXT' | customTranslate | translate}}
+                :
+              </div>
+              <div *ngIf="consent.status === 'REVOKED'">
+                {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.REVOKED_TEXT' | customTranslate | translate}}
+                :
+              </div>
+              <div *ngIf="consent.status === 'UNSIGNED'">
+                {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.UNSIGNED_TEXT' | customTranslate | translate}}
+                :
+              </div>
+              <ul>
+                <li *ngFor="let attribute of consent.attributes">
+                  {{ attribute.displayName }}
+                </li>
+              </ul>
+              <div class="d-flex">
+                <button
+                  *ngIf="consent.status === 'GRANTED'"
+                  mat-flat-button
+                  color="warn"
+                  (click)="rejectConsent.emit(consent.id)">
+                  {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.REVOKE_CONSENT_BUTTON' | customTranslate | translate}}
+                </button>
+                <button
+                  *ngIf="consent.status === 'UNSIGNED'"
+                  mat-flat-button
+                  class="mr-2"
+                  color="warn"
+                  (click)="rejectConsent.emit(consent.id)">
+                  {{'SHARED_LIB.PERUN.COMPONENTS.CONSENTS_LIST.REJECT_CONSENT_BUTTON' | customTranslate | translate}}
+                </button>
```

Comment:
First two buttons are the same jsut different text. Use one button with condition on the text.